x/config/socks5.go

```diff
@@ -27,18 +27,46 @@ func wrapStreamDialerWithSOCKS5(innerSD func() (transport.StreamDialer, error),
 		return nil, err
 	}
 	endpoint := transport.StreamDialerEndpoint{Dialer: sd, Address: configURL.Host}
-	dialer, err := socks5.NewStreamDialer(&endpoint)
+	client, err := socks5.NewClient(&endpoint)
 	if err != nil {
 		return nil, err
 	}
 	userInfo := configURL.User
 	if userInfo != nil {
 		username := userInfo.Username()
 		password, _ := userInfo.Password()
-		err := dialer.SetCredentials([]byte(username), []byte(password))
+		err := client.SetCredentials([]byte(username), []byte(password))
 		if err != nil {
 			return nil, err
 		}
 	}
-	return dialer, nil
+
+	return client, nil
+}
+
+func wrapPacketDialerWithSOCKS5(_ func() (transport.StreamDialer, error), innerPD func() (transport.PacketDialer, error), configURL *url.URL) (transport.PacketDialer, error) {
+	// Use a TCP dialer to connect to the SOCKS5 server.
+	sd := &transport.TCPDialer{}
```

Comment:
@fortuna my main concern with using `innerSD` is that since this only used for UDP association request and if the base dialer is something other than plain `&transport.TCPDialer{}` (for example if base dialer is shadowsocks) it can cause problems.

Reply:
Why would it cause problems? Without using the innerSD you are not properly composing the transports. You can't do multi-hop without that.

Reply:
@fortuna nevermind. You are right since we are not transferring data using innerSD and we just use it to dial to socks5 server and send connect/associate command. I applied the change.